Src/FluentAssertions/Xml/Equivalency/XmlReaderValidator.cs

```diff
@@ -38,6 +40,21 @@ public void Validate(bool shouldBeEquivalent)
 
         private Failure Validate()
         {
+            if (subjectReader is null && expectationReader is null)
```

Comment:
🤔 It's kind of weird that this block and `ValidateAgainstNull` are both testing against `null`. Shouldn't this be combined?

Reply:
🤡 I Just copied it from https://github.com/fluentassertions/fluentassertions/blob/develop/Src/FluentAssertions/Primitives/StringValidator.cs

Reply:
You can't combine it, as it is a tri-state:

1. Both `null` -> Stop validation without failure
2. One `null` -> Stop validation with failure
3. None `null` -> Continue validation

Reply:
Fair enough